src/LayoutManager/LayoutManager.ts

```diff
@@ -130,11 +142,27 @@ export class LayoutManager {
     const { target } = context;
     const { canvas } = target;
     // handle layout triggers subscription
+    // @TODO: gate the registration when the group is interactive
     if (
       context.type === LAYOUT_TYPE_INITIALIZATION ||
       context.type === LAYOUT_TYPE_ADDED
     ) {
-      this.subscribeTargets(context);
+      const withDifferentParent = context.targets.filter((object) =>
+        this.subscribe(object, context)
+      );
+      // if objects have a different parent from group, they are in an active seleciton
+      // so we need to subscribe the active selection event to trigger the parent performLayout
+      withDifferentParent.forEach(({ group, parent }) => {
+        if (parent) {
+          // we may subscribe an active selection more than once,
+          // on each parent's layout manager.
+          // each parent's can subscribe itself just once because of unsubscribe
+          parent.layoutManager.subscribe(group!, {
+            ...context,
+            target: parent,
+          });
+        }
+      });
```

Comment:
@shaman this is fix 1.

Reply:
I think we may want to subclass an ActiveSelectionLayoutManager for this, what do you think?

Reply:
OR we can subscribe all groups to the canvas level events and check if the target is the active selection or a child.
However it means we need the canvas ref to begin with and we don't in the initialization trigger

Reply:
So a different approach would be to have an ActiveSelectionLayoutManager and have it trigger layout on the parents when it triggers its own layout.

Reply:
I can check if it creates overall less code.
If it creates more code, no.

Reply:
I ll try to test this before tomorrow so we can check the end result together